.github/workflows/compilation.yml

```diff
@@ -0,0 +1,50 @@
+name: CI-compile
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    # container: ghcr.io/${{ github.repository_owner }}/mipsr5900el-gentoo-linux-gnu:main
+    container: ghcr.io/AKuHAK/mipsr5900el-gentoo-linux-gnu:main
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Compile project
+        env:
+          INSTALL_MOD_PATH: ../initramfs/ps2/
+          INSTALL_MOD_STRIP: 1
+        run: |
+          mv -rf ../svr/iniramfs ../
+          make -j $(getconf _NPROCESSORS_ONLN) ps2_defconfig
+          make -j $(getconf _NPROCESSORS_ONLN) oldconfig
+          make -j $(getconf _NPROCESSORS_ONLN) vmlinux
+          make -j $(getconf _NPROCESSORS_ONLN) modules
+          make -j $(getconf _NPROCESSORS_ONLN) modules_install
+          make -j $(getconf _NPROCESSORS_ONLN) vmlinuz
```

Comment:
Aren't these steps done in

https://github.com/frno7/mipsr5900el-gentoo-linux-gnu/blob/9e01f44b0d5e04a496c8ac27e13494bf208eb63a/Dockerfile#L49-L55

already? I agree that they're better placed here, but then I suppose we should remove them from the mipsr5900el-gentoo-linux-gnu repo? Does it make sense to make the Dockers more modular, so that QEMU would be built only in the [qemu](https://github.com/frno7/qemu) repo, and so on?

Reply:
This is a big philosophical question :D Docker image currently is so unoptimized, that it is almost 3.5Gb in size and 1.7Gb reserved only for kernel sources. I made a Docker image to hold All-In-One inside one container, so everyone can download it and make a fast kernel compilation on his machine just by typing `cd /srv/kernel; make vmlinuz`. As about this Github Action, I agree that it is doing "double work", but I don't see that this is a problem. Docker contains Linux kernel from the past, but Github Action compiles an actively developed branch that can be more actual.
Anyway, I don't think that this action should be affected by this. You probably should create an issue in the Docker container repository for possible removing kernel sources from the container.

> Does it make sense to make the Dockers more modular, so that QEMU would be built only in the [qemu](https://github.com/frno7/qemu) repo, and so on?

It can make sense, but it is difficult to calculate necessary dependencies. It is much easier to hold everything inside an all-in-one container than to manage dependency calculations for each repository. This container can be overkill for most repoes, but it is much easier to maintain. And it will be faster to complete cause all utilities are already precompiled inside the container.
Maybe only the kernel itself can be removed from container, cause it is very big and probably will not be dpendent for other things (except of initramfs of course, I mainly included it there for easier intiramfs managing).

Reply:
Right, let's celebrate this package achievement now, and tune things later as we go along. Will you push the branch name thing mentioned in https://github.com/frno7/linux/pull/69#discussion_r854048840 before we merge this? With the https://github.com/frno7/mipsr5900el-gentoo-linux-gnu/commit/9e01f44b0d5e04a496c8ac27e13494bf208eb63a fix in, chances are that it'll actually boot nice and proper. :-)

Reply:
Done

Reply:
Hmm... [Initialize containers](https://github.com/AKuHAK/linux/runs/6099710475?check_suite_focus=true) fails with `invalid reference format: repository name must be lowercase`. It seems `github.ref` evaluates to `refs/heads/ps2-main`, which having `/` won’t work as a branch name. We’d need to get rid of the `refs/heads/` prefix. A need for a substring function, again. :-)

[Environment files](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files) seems to offer a solution. Adapting its example, something like

```
steps:
  - name: Obtain file tag from branch and commit
    id: obtain_file_tag
    run: |
      echo "FILE_TAG=${GITHUB_REF#refs/*/}-${GITHUB_SHA:0:8}" >> $GITHUB_ENV
  - name: Echo file tag
    id: echo_file_tag
    run: |
      echo "${{ env.FILE_TAG }}" # Should output "ps2-main-dc19f33e" or suchlike
```

where `${GITHUB_REF#refs/*/}` (_shortest matching pattern deleted_) and `${GITHUB_SHA:0:8}` (_substring expansion_) are [Bash parameter expansion](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html) rules.

Reply:
According to the documentation for `GITHUB_REF` it’s

> The branch or tag ref that triggered the workflow run. For branches this is the format `refs/heads/<branch_name>`, for tags it is `refs/tags/<tag_name>`, and for pull requests it is `refs/pull/<pr_number>/merge`. This variable is only set if a branch or tag is available for the event type. For example, `refs/heads/feature-branch-1`.

A problematic case will be pull requests with `refs/pull/<pr_number>/merge` having triple `/`. We could use `${GITHUB_REF##refs/*/}` with `##` serving as the _longest matching pattern deleted_ rather than the _shortest_ with a single `#`. It’ll evaluate `refs/pull/17/merge` to `merge`, rather than `17/merge`, which wouldn’t work with `/`. Another, more useful alternative, would be to have FILE_TAG become `pull-17`, `17-merge`, or `pull-17-merge`. :-)